readers/multishard.cc

```diff
@@ -607,7 +607,7 @@ future<> evictable_reader_v2::fill_buffer() {
         // First make sure we've made progress w.r.t. _next_position_in_partition.
         // This loop becomes inifinite when next pos is a partition start.
         // In that case progress is guranteed anyway, so skip this loop entirely.
-        while (!_next_position_in_partition.is_partition_start() && next_mf && _tri_cmp(_next_position_in_partition, buffer().back().position()) <= 0) {
+        while (!_next_position_in_partition.is_partition_start() && next_mf && _tri_cmp(buffer().back().position(), _next_position_in_partition) <= 0) {
             push_mutation_fragment(_reader->pop_mutation_fragment());
             next_mf = co_await _reader->peek();
         }
```

Comment:
I think we can drop this entire loop. We already validated in `examine_first_fragments()` that the first non-dropped fragment in the new buffer is` >= _next_position_in_partition`. So this loop is pointless now that I think of it.
We ensure separately that we have not stopped in the middle of a range of fragments that have the same position (the loop below).

Reply:
> I think we can drop this entire loop. We already validated in examine_first_fragments() that the first non-dropped fragment in the new buffer is >= _next_position_in_partition.

But only in the case of `_reader_recreated`, no?